### PR TITLE
OCRVS-2298: Introduced New Config microservice

### PR DESCRIPTION
**Pull request by yeasinhossain-dsi**

### CODE REVIEW

File: packages/client/src/offline/selectors.ts [hunk 1]
```diff
@@ -33,9 +34,12 @@ export function isOfflineDataLoaded(
     state.forms.userForm &&
     state.languages &&
     state.assets &&
-    state.templates
+    state.templates &&
+    state.config
 
-  return Boolean(hasAllRequiredData)
+  const isOfflineDataLoaded = Boolean(hasAllRequiredData)
+  if (isOfflineDataLoaded) merge(window.config, state.config)
```
Comment (rikukissa) [inline packages/client/src/offline/selectors.ts:41]: Would there be another way of doing this? Mutating the global `window.config` inside a call to a previously pure function feels a bit misleading from code reader point of view

File: packages/config/src/handlers/applicationConfigHandler.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+export default async function applicationHandler(
+  request: Hapi.Request,
+  h: Hapi.ResponseToolkit
+) {
+  try {
+    // tslint:disable-next-line
+    const appConfig: IApplicationConfigurationModel | null =
+      await ApplicationConfig.findOne({})
+    return appConfig || 'no configuration'
```
Comment (rikukissa) [inline packages/config/src/handlers/applicationConfigHandler.ts:26]: Would returning an empty body and a 404 status work if no config exists?

File: packages/config/src/handlers/applicationConfigHandler.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  } catch (ex) {
+    logger.error(ex)
+    return {}
```
Comment (rikukissa) [inline packages/config/src/handlers/applicationConfigHandler.ts:29]: Here we could maybe return a 500 status

File: packages/client/src/offline/selectors.ts (hunk 1 above)
Comment (euanmillar) [reply packages/client/src/offline/selectors.ts:41]: I think it is OK.  By merging state.config into window.config, the props loaded from the config microservice become globally accessible and then there is no need to refactor other parts of the code to retrieve config from redux state

**Pull request merged**